api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -204,8 +204,9 @@ URI getResolvedImpersonationTokenEndpoint() {
    */
   @Value.Lazy
   Optional<HttpAuthentication> getBasicAuthentication() {
-    return getClientSecret()
-        .map(s -> BasicAuthenticationProvider.create(getClientId(), s.getString()));
+    return getClientSecretSupplier()
+        .map(Supplier::get)
+        .map(s -> BasicAuthenticationProvider.create(getClientId(), s));
```

Comment:
Sorry for the late review. If the intention of the `getClientSecretSupplier()` to to dynamically resolve the secret (potentially return a different value on every call), should we not call it from inside the `BasicAuthentication` when it is actually needed for HTTP calls?

Reply:
True, I will open another PR.